api/src/main/java/me/lokka30/treasury/api/economy/EconomyProvider.java

```diff
@@ -196,6 +272,38 @@ void createAccount(
      */
     void retrieveNonPlayerAccountIds(@NotNull EconomySubscriber<Collection<String>> subscription);
 
+    /**
+     * {@link #retrieveAccountIds(EconomySubscriber)}
+     */
+    @NotNull
+    default CompletableFuture<Collection<String>> retrieveNonPlayerAccountIds() {
+        return EconomySubscriber.asFuture(this::retrieveNonPlayerAccountIds);
+    }
+
+    /**
+     * Request all {@link NonPlayerAccount non player accounts} the given player is a member of.
+     *
+     * @param playerId the player
+     * @return future with data
+     */
+    @NotNull
+    default CompletableFuture<Collection<String>> retrieveAllAccountsPlayerIsMemberOf(@NotNull UUID playerId) {
+        Objects.requireNonNull(playerId, "playerId");
+        return this.retrieveAccountIds().thenComposeAsync(identifiers -> {
+            if (identifiers.isEmpty()) {
+                return CompletableFuture.completedFuture(Collections.emptySet());
+            }
+            Collection<CompletableFuture<Account>> futures = new HashSet<>();
+            for (String identifier : identifiers) {
+                futures.add(this.retrieveAccount(identifier));
+            }
+            return FutureHelper.mjf(account -> account.isMember(playerId),
+                    Account::getIdentifier,
+                    futures
+            );
```

Comment:
`FutureHelper#mjf` uses [`CompletableFuture#join()`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html#join--) which
> throws an (unchecked) exception if completed exceptionally

but this excepcion is not handled in the `mjf` method or this method, nor is it specifyed in the documentation that it's not handled